### PR TITLE
Create CommentsController Rspec

### PR DESCRIPTION
**Pull request by ShahirMansuri786**

### CODE REVIEW

File: app/views/comments/new.html.erb [hunk 1]
```diff
@@ -1,33 +1,29 @@
 <html >
   <head>
-  
-     <style>
-     .con:hover{
-      box-shadow: 10px 10px 20px 10px grey ;}
+    <style>
+      .con:hover{
+        box-shadow: 10px 10px 20px 10px grey ;}
```
Comment (Sahilchouhan20) [inline app/views/comments/new.html.erb:5]: Write css in stylesheet

File: spec/controllers/comments_controller_spec.rb [hunk 2]
```diff
@@ -5,22 +5,27 @@
...
 		end
   end
 
-	describe "GET #new" do
+	describe "CommentsController#new" do
 		it "returns a new instance for new comment" do
 			comment = FactoryBot.create(:comment , post: post)
-			get :new
+			get new_post_comment_path(post)
+			expect(assigns(:post_comment)).to be_a_new(Comment)
```
Comment (Sahilchouhan20) [inline spec/controllers/comments_controller_spec.rb:26]: Follow indentation while writing code.

File: spec/requests/comments_spec.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      end
+    end
+  end
+
+	describe "GET /posts/:post_id/comments/:id" do
+		it "comments#show" do
+			comment = FactoryBot.create(:comment , post: post)
+			get "/posts/#{comment.post_id}/comments/#{comment.id}"
+			expect(response).to have_http_status(:success)
+			expect(assigns(:post_comment)).to eq(comment)
```
Comment (aadi163) [inline spec/requests/comments_spec.rb:34]: Correct indent.

File: spec/requests/comments_spec.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		end
+	end
+
+
+  describe "Post /posts/:post_id/comments" do
+
```
Comment (aadi163) [inline spec/requests/comments_spec.rb:40]: Why did you give extra spaces.

File: app/views/comments/new.html.erb (hunk 1 above)
Comment (ShahirMansuri786) [reply app/views/comments/new.html.erb:5]: I didn't use too many css so  that's why i have used here

File: spec/requests/comments_spec.rb [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    it "creates a new comment and redirects to the post" do
+      post = FactoryBot.create(:post )
+      user = FactoryBot.create(:user) 
+
+      comment_attributes = FactoryBot.attributes_for(:comment, user_id: user.id)
+      # comment = FactoryBot.create(:comment, user_id: user.id )
+      # post post_comments_path(post.id) , params: { comment: {comment_text: "Nice" , user_id: user.id} }
+      
+      process :create, method: :post ,  params: (comment_attributes) 
+
```
Comment (aadi163) [inline spec/requests/comments_spec.rb:50]: Here as well you gave extra space.

File: spec/controllers/comments_controller_spec.rb (hunk 2 above)
Comment (ShahirMansuri786) [reply spec/controllers/comments_controller_spec.rb:26]: sure

File: spec/requests/comments_spec.rb [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    context "when not signed in" do
+      before { sign_out user }
+      it "redirects to the sign-in page" do
+        delete  "/posts/#{post.id}/comments/#{comment.id}"
+        expect(response).to redirect_to(user_session_path)
+      end
+    end
+  end
+
```
Comment (aadi163) [inline spec/requests/comments_spec.rb:76]: here also

File: spec/requests/comments_spec.rb (hunk 3 above)
Comment (ShahirMansuri786) [reply spec/requests/comments_spec.rb:34]: ok sure

File: spec/requests/comments_spec.rb (hunk 4 above)
Comment (ShahirMansuri786) [reply spec/requests/comments_spec.rb:40]: ok I will remove it

File: spec/requests/comments_spec.rb (hunk 5 above)
Comment (ShahirMansuri786) [reply spec/requests/comments_spec.rb:50]: I have gave spaces for code better undestanding here

File: spec/requests/comments_spec.rb (hunk 6 above)
Comment (ShahirMansuri786) [reply spec/requests/comments_spec.rb:76]: I have gave spaces for code better undestanding here

**Pull request merged**